fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -2909,7 +2918,29 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
         LOG.info("Successfully create materialized view[{};{}]", mvName, mvId);
 
         // NOTE: The materialized view  has been added to the database, and the following procedure cannot throw exception.
-        // todo add async task after task framework is completed
+        if (createMvSuccess) {
+            if (materializedView.getRefreshScheme().getType() == MaterializedView.RefreshType.ASYNC) {
+                // create task
+                Task task = new Task();
+                long taskId = GlobalStateMgr.getCurrentState().getNextId();
+                task.setId(taskId);
+                task.setName("mv-" + materializedView.getId());
+                task.setCreateTime(System.currentTimeMillis());
+                task.setDbName(getDb(materializedView.getDbId()).getFullName());
+                Map<String, String> taskProperties = Maps.newHashMap();
+                taskProperties.put("mvId", String.valueOf(materializedView.getId()));
+                task.setProperties(taskProperties);
+                task.setDefinition(materializedView.getViewDefineSql());
+                task.setExpireTime(0L);
```

Comment:
Nerver expire?

Reply:
Yes, 0 means never expires. The Task framework has the logic of automatic cleanup after expiration, so this requires the materialized view to maintain the creation and drop of the Task itself.